helper/keysutil/policy.go

```diff
@@ -959,12 +969,19 @@ func (p *Policy) Sign(ver int, context, input []byte, algorithm string) (*Signin
 	// Convert to base64
 	encoded := base64.StdEncoding.EncodeToString(sig)
 
-	template := p.VersionTemplate
-	if template == "" {
-		template = DefaultVersionTemplate
-	}
+	var prefix string
+	prefixRaw, ok := p.versionPrefixCache.Load(ver)
+	if !ok {
+		template := p.VersionTemplate
+		if template == "" {
+			template = DefaultVersionTemplate
+		}
 
-	prefix := strings.Replace(template, "{{version}}", strconv.Itoa(ver), -1)
+		prefix = strings.Replace(template, "{{version}}", strconv.Itoa(ver), -1)
+		p.versionPrefixCache.Store(ver, prefix)
+	} else {
+		prefix = prefixRaw.(string)
```

Comment:
I don't see code for Decryption/Verification. Those need to parse the version out of the templated output, I would assume they need some lookup logic.

Reply:
I think the best we can do here is cache the outcome of `strings.Split(template, "{{version}}")`, will add